tests/python/rest_api/test_jobs.py

```diff
@@ -600,6 +611,8 @@ def test_can_get_gt_job_meta(self, admin_user, tasks, task_mode):
         with make_api_client(user) as api_client:
             (gt_job_meta, _) = api_client.jobs_api.retrieve_data_meta(gt_job.id)
 
+        request.addfinalizer(lambda: self.delete_gt_job(user, gt_job.id))
```

Comment:
I think it's better to put right after the `_get_or_create_gt_job()` call in each case.

Reply:
Some functions later on fetch the metadata of this newly created gt_job. So, this change might cause problem.

Reply:
Finalizers are called on the test tear down. Which potential problems do you see?

Reply:
Ohh is that so, sorry I thought they are called in the order they are called.